### DISCUSSION TITLE
How can I enable the panning on iPad.

### DISCUSSION THREAD

**Discussion by HalilFocic** [maxGraph/maxGraph#562] [Q&A]
Currently I use the right click on web to navigate around graph. But on mobile I wasn't able to quite the this to work. I also tried by enabling the paning using left mouse button in hopes that would change but with no success.

**Comment by tbouffard** [comment]
Hi
there are some known issue with touch/gestures events. See #62 and #149.
Touch/gestures events are known to better work on iOS than on Adroid. The issues are mainly with the zoom.

Panning is supposed to work, at least a bit.
Can you please share an example where you see the problem you describe? A lot of stories enable panning in https://maxgraph.github.io/maxGraph/demo/
You can also test with https://maxgraph.github.io/maxgraph-integration-examples/vitejs-ts/ or locally with https://github.com/maxGraph/maxGraph/tree/v0.13.1/packages/ts-example

**Reply by HalilFocic** [reply]
Hello. 
I tried both links and they are kinda broken on my iPad. I on most stories I was able to use right click to scroll through graph without moving the borders(if you get me). Basically I was able to move the "Hello" box outside of my viewport which i wasn't quite able to reproduce on my iPad. I did in developer console that in every example I saw that i had error "Invalid x supplied" and in call stack I saw function "isScrollEvent" which is inside graph view.

isScrollEvent function tries to get the x and y but using evt.clientX and evt.clientY which I think is what is causing the issue since clientX and clientY are not avaiable directly inside TouchEvent like they are inside MouseEvent. I will look into this on the weekend in my fork and try to make it work better since I have few android and ios  phones and tablets that I can test my theory on. If I am able to atleast make this behaviour functionable on my tablets I will continue in discussions about how we can work together have support for all devices. Cheers

**Comment by HalilFocic** [comment]
Changing the code inside scrollEvent fixed my problem

**Reply by tbouffard** [reply]
Thanks, I have seen you created #566 🎉 
I will have a look at the PR and hope to merge it quickly!
